File: mutuelle/src/app/model/membre.ts
import { Agence } from "./agence";
import { Avance } from "./avance";
import { CotisationList } from "./cotisationList";
import { Credit } from "./credit";
import { IMembreBase } from "./imembrebase";
import { Service } from "./service";
import { Sexe } from "./sexe";


export class Membre {
  id?: number;
  nom?: string;
  prenom?: string;
  sexeId?: number;
  sexe?: Sexe;
  photo?: string;
  agenceId?: number;
  agence?: Agence;
  serviceId?: number;
  service?: Service;
  estActif?: boolean;
  dateAdhesion?: Date;
  fraisAdhesion?: number;
  telephone?: string;
  email?: string;
  cotisations?: CotisationList[];
  avances?: Avance[];
  credits?: Credit[];
}

File: WebApi/Models/Membre.cs
namespace WebApi.Models
{
    public class Membre : BaseEntity
    {
        public string Nom { get; set; } = string.Empty;
        public string Prenom { get; set; } = string.Empty;
        public int SexeId { get; set; }
        public Sexe Sexe { get; set; } = new Sexe{};
        public string? Photo { get; set; }
        public int AgenceId { get; set; }
        public Agence Agence { get; set; } = new Agence{};
        public int ServiceId { get; set; }
        public Service Service { get; set; } = new Service{};
        public bool EstActif { get; set; }
        public string Telephone { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public ICollection<Cotisation> Cotisations { get; set; } = new List<Cotisation>();
        public ICollection<Avance> Avances { get; set; } = new List<Avance>();
        public ICollection<Credit> Credits { get; set; } = new List<Credit>();
    }
}